### PR TITLE
Created API token docs

### PR DESCRIPTION
**Pull request by lampy2**
Add explanation on how to generate an api token (#1398)

Signed-off-by: Gerwin Lammers <gerwinlammers@gmail.com>

### CODE REVIEW

**Comment by lampy2** [comment]
Hi @Confectrician, sorry I messed up with trying to recover yesterday. Did some more reading on all the therms and how Git is working and I start to understand it better (hopefully:))
At least there is only one commit now.

**Comment by Confectrician** [comment]
Hey @lampy2 thanks for keeping up and being interested. 👍 

Looks promising so far.
Only 1 commit, dco check working.

I would do a review, because i have seen some things that would need changes.
We can then mess up this pr too. 😛 

> sorry I messed up with trying to recover yesterday.

Sometimes messing up and learning for the future is faster than reading the docs. 🙂

File: configuration/apitokens.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+layout: documentation
+title: openHAB API tokens
+---
+
+{% include base.html %}
+
+# openHAB API tokens
+
+API tokens are an authentication method, like an username and password. Usually you would use an API token to authenticate an external service or script.
```
Comment (Confectrician) [inline configuration/apitokens.md:10]: Please use one line per sentence.

File: configuration/apitokens.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Generate an API token
+Sign in to openHAB with your administrator username and password, by clicking on the sign-in button (bottem left)
+
+After signing in, click again on the profile button to access your profile page. Click on Create new API token.
```
Comment (Confectrician) [inline configuration/apitokens.md:15]: Please use one line per sentence.

File: configuration/apitokens.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+![apitoken login](images/apitoken_login.png)
+
+Use your admin username and password and fill in the token name (e.g. the service you want to gain access for) and a Token Scope.
+
+![Create token](images/apitoken_create_token.png)
+
+Now the token is created. Copy the complete token.
```
Comment (Confectrician) [inline configuration/apitokens.md:23]: Please use one line per sentence.

File: configuration/apitokens.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+![Token](images/apitoken_token.png)
+
+This token is in the used service your username, leave the password empty.
+As example, below in NodeRed configuration the generated token is used as username.
+
+![Example](images/apitoken_example.png)
+
+## Overview of generated tokens
+In your profile overview page you can find an overview of generated tokens. Additionally, they can be deleted here.
```
Comment (Confectrician) [inline configuration/apitokens.md:33]: Please use one line per sentence.

File: configuration/apitokens.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+layout: documentation
+title: openHAB API tokens
+---
+
+{% include base.html %}
```
Comment (Confectrician) [inline configuration/apitokens.md:6]: You can safely remove this line.
I have to remove all others too.
It is not needed for our current website builds.

**Comment by lampy2** [comment]
> Hey @lampy2 thanks for keeping up and being interested. 👍

Didn't expect it to be easy for the first time, but there are really a lot of small steps to go right to make it happen. But nice to learn and hopefully I can use it more often ;) 

> Sometimes messing up and learning for the future is faster than reading the docs. 🙂

Yes indeed, cycling you also didn't learned by reading a book about cycling.  🙂

> I would do a review, because i have seen some things that would need changes.
> We can then mess up this pr too. 😛

To avoid messing up to much again, I read there are multiple options to update the PR. Can you advice which one to use?

- New commit to `lampy2:apitoken-create` and it will be added automatically as second commit to this PR. But then there are 2 commits in this PR.
- use the `amend` function, but this should not be the preferred why right?

**Comment by Confectrician** [comment]
So your dco check is `green|valid` currently.
When you have configured git properly on your pc and add another commit in vscode 
through console with `git commit -s -m "My commit message"`, the check chould stay valid.

It just validates every commit for the sign off message at the end.
No need for amending, then.

File: configuration/apitokens.md (hunk 5 above)
Comment (lampy2) [reply configuration/apitokens.md:6]: Ok! Indeed copied it from another page.

**Comment by Confectrician** [comment]
Seems that everything worked, congrats. 👌

Are you satisfied with the result in the preview page?

From my side it looks good and we have to do one last step.
Add the page to our sidebar navigation config, to get it presented in the sidebar.
(I will add it to your Pull Request myself.)

**Comment by lampy2** [comment]
Thanks! Not to bad for a first time, but a good mentor helps a lot! 😀 
Yes pretty satisfied, one of the pictures seems to be a bit large compared to the text, but is this determined by the file size itself? Or can you scale it later as well?

**Comment by Confectrician** [comment]
I am not aware of being able to resize images here.
(Though it should be possible with Markdown theoretically, we do not use it at all.)

**Review by Confectrician — APPROVED** [review]
Thanks 👍

Good work and progress for a first time contribution. 🙂

**Comment by ghys** [comment]
Indeed. Good to have this documented.

Regarding screenshots, I think we should introduce some kind of guidance on how to produce them for consistency's sake.
For instance, desktop theme, light, filled bars, 720p (you can use the [Device Toolbar](https://developers.google.com/web/tools/chrome-devtools/device-mode) on Chrome & others, and even take the screenshot from there):

image

**Comment by Confectrician** [comment]
Maybe something for
https://community.openhab.org/t/wiki-how-to-contribute-to-the-openhab-documentation/111308?u=confectrician

**Pull request merged**